src/Extractor/ContentExtractor.php

```diff
@@ -660,6 +639,65 @@ function ($element, $currentEntity) {
         return $this->success;
     }
 
+    /**
+     * Process string replacements in the $html body.
+     *
+     * @param string     $html
+     * @param string     $url
+     * @param SiteConfig $siteConfig Will avoid to recalculate the site config
+     *
+     * @return string $html with replacements performed
+     */
+    public function processStringReplacements(string $html, string $url, ?SiteConfig $siteConfig = null)
+    {
+        /* We repeat this step from process(), so this method can be called on its own */
+        $this->prepareSiteConfig($html, $url, $siteConfig);
+
+        if (!empty($this->siteConfig->find_string)) {
+            if (\count($this->siteConfig->find_string) === \count($this->siteConfig->replace_string)) {
+                $html = str_replace($this->siteConfig->find_string, $this->siteConfig->replace_string, $html, $count);
+                $this->logger->info('Strings replaced: {count} (find_string and/or replace_string)', ['count' => $count]);
+            } else {
+                $this->logger->info('Skipped string replacement - incorrect number of find-replace strings in site config');
+            }
+            unset($count);
+        }
+        $this->logger->debug('HTML after site config strings replacements', ['html' => $html]);
+
+        return $html;
+    }
+
+    /**
+     * Set and prepare the SiteConfig, or get a default.
+     * If a siteConfig is already set and no prepare site config is passed, this is a noop.
+     *
+     * @param string     $html
+     * @param string     $url
+     * @param SiteConfig $siteConfig Will avoid to recalculate the site config
+     *
+     * @return string $html with replacements performed
+     */
+    private function prepareSiteConfig($html, $url, ?SiteConfig $siteConfig = null) {
+        if (null !== $this->siteConfig && null === $siteConfig) {
+            return;
+        }
+        $this->siteConfig = $siteConfig;
+        if (null === $this->siteConfig) {
+            $this->siteConfig = $this->buildSiteConfig($url, $html);
+        }
+
+        // add lazyload information from siteconfig
+        if ($this->siteConfig->src_lazy_load_attr &&
+            !\in_array($this->siteConfig->src_lazy_load_attr, $this->config['src_lazy_load_attributes'], true)
+        ) {
+            /* XXX: Could this become an issue (creating duplicates) if we passed the same $siteConfig multiple times? */
```

Comment:
```suggestion
```
Maybe, but I don't think it can generates error if we have duplicate information inside it